Make sure the icalendar buffer is unfolded before processing itIndex: devel/notmuch/emacs/notmuch-show.el
===================================================================
--- devel.orig/notmuch/emacs/notmuch-show.el	2012-09-17 11:28:24.484941490 +0200
+++ devel/notmuch/emacs/notmuch-show.el	2013-01-08 13:28:11.895658702 +0100
@@ -740,18 +740,23 @@
   (notmuch-show-insert-part-header nth declared-type content-type (plist-get part :filename))
   (insert (with-temp-buffer
 	    (insert (notmuch-get-bodypart-content msg part nth notmuch-show-process-crypto))
-	    (goto-char (point-min))
-	    (let ((file (make-temp-file "notmuch-ical"))
-		  result)
-	      (icalendar--convert-ical-to-diary
-	       (icalendar--read-element nil nil)
-	       file t)
-	      (set-buffer (get-file-buffer file))
-	      (setq result (buffer-substring (point-min) (point-max)))
-	      (set-buffer-modified-p nil)
-	      (kill-buffer (current-buffer))
-	      (delete-file file)
-	      result)))
+	    (let (
+		  (unfolded-buffer (icalendar--get-unfolded-buffer (current-buffer)))
+		  )
+	      (with-current-buffer unfolded-buffer
+		(goto-char (point-min))
+		(let ((file (make-temp-file "notmuch-ical"))
+		      result)
+		  (icalendar--convert-ical-to-diary
+		   (icalendar--read-element nil nil)
+		   file t)
+		  (set-buffer (get-file-buffer file))
+		  (setq result (buffer-substring (point-min) (point-max)))
+		  (set-buffer-modified-p nil)
+		  (kill-buffer (current-buffer))
+		  (delete-file file)
+		  result)))
+	    ))
   t)
 
 ;; For backwards compatibility.
